spec/jobs/notify_email_job_spec.rb

```diff
@@ -0,0 +1,172 @@
+    let(:govuk_notify_api_key) { nil }
+
+    it 'raises an NoMethodError' do
+      expect { perform! }.to raise_error(NoMethodError, /undefined method `length' for nil/)
+    end
+  end
+
+  context 'when GOVUK_NOTIFY_TEMPLATE_ID env var is not set' do
+    let(:govuk_notify_template_id) { nil }
+
+    it 'raises an ArgumentError' do
+      expect { perform! }.to raise_error(ArgumentError, /Missing template ID/)
+    end
+  end
+
+  context 'when notification and subscription are active' do
+    context 'when the notification has already been delivered' do
+      # NB: need to be explicit about the time in the test otherwise there can be a few nanoseconds difference in CircleCI tests
+      let(:delivered_at) { DateTime.parse('2020-02-02 02:02:02.00') }
+
+      before { allow(MoveMailer).to receive(:notify) }
+
+      it 'does not notify the mailer' do
+        perform!
+        expect(MoveMailer).not_to have_received(:notify)
+      end
+
+      it 'returns nil' do
+        expect(perform!).to be nil
+      end
+
+      it 'does not set delivered_at' do
+        expect { perform_ignore_errors! }.not_to change { notification.reload.delivered_at }.from(delivered_at)
+      end
+
+      it 'does not update delivery_attempted_at' do
+        expect { perform_ignore_errors! }.not_to change { notification.reload.delivery_attempted_at }.from(nil)
+      end
+
+      it 'does not update delivery_attempts' do
+        expect { perform_ignore_errors! }.not_to change { notification.reload.delivery_attempts }.from(0)
+      end
+
+      it 'does not update response_id' do
+        expect { perform_ignore_errors! }.not_to change { notification.reload.response_id }.from(nil)
+      end
+    end
+
+    context 'when the email is successfully delivered to Gov.uk Notify' do
+      let(:notify_response) {
+        instance_double(ActionMailer::MessageDelivery, deliver_now!:
+            instance_double(Mail::Message, govuk_notify_response:
+                instance_double(Notifications::Client::ResponseNotification, id: response_id)))
+      }
+      let(:response_id) { SecureRandom.uuid }
+
+      before { allow(MoveMailer).to receive(:notify).and_return(notify_response) }
+
+      it 'notifies the mailer' do
+        perform!
+        expect(MoveMailer).to have_received(:notify)
+      end
+
+      it 'updates delivered_at' do
+        expect { perform! }.to change { notification.reload.delivered_at }.from(nil)
+      end
+
+      it 'updates delivery_attempted_at' do
+        expect { perform! }.to change { notification.reload.delivery_attempted_at }.from(nil)
+      end
+
+      it 'updates delivery_attempts' do
+        expect { perform! }.to change { notification.reload.delivery_attempts }.from(0).to(1)
+      end
+
+      it 'updates response_id' do
+        expect { perform! }.to change { notification.reload.response_id }.from(nil).to(response_id)
+      end
+    end
+
+    context 'when Gov.uk Notify does not respond as expected' do
```

Comment:
maybe we can be more explicit about  `...does not respond as expected`

Reply:
ok, have renamed it to: `when Gov.uk Notify does not respond as with a govuk_notify_response object`